src/mechanics/input.rs

```diff
@@ -1,628 +1,594 @@
+//            }
+//        }
+//
+//        let tile_side_length = 64.0;
+//        let projected_position = entity_transform.translation + direction;
+//
+//        for &collision_tile in collision_tiles.iter() {
+//            let tile_position = Vec3::new(
+//                collision_tile.px.x as f32,
+//                (level_dimension.height as i32 - (collision_tile.px.y)) as f32,
+//                0.0,
+//            );
+//
+//            let projected_dimensions = Vec2::new(tile_side_length, tile_side_length);
+//            let tile_dimensions =
+//                Vec2::new(collision_tile.width as f32, collision_tile.height as f32);
+//
+//            let has_collided =
+//                Aabb2d::new(projected_position.truncate(), projected_dimensions / 2.0).intersects(
+//                    &Aabb2d::new(tile_position.truncate(), tile_dimensions / 2.0),
+//                );
+//
+//            if has_collided {
+//                entity_movement_broadcast.send(PlayerMovementActions::Bumping);
+//                *moving = MovementIntent::Idle;
+//                return;
+//            }
+//        }
+//
+//        entity_transform.translation = projected_position;
+//        entity_movement_broadcast.send(PlayerMovementActions::Walking);
+//        *moving = MovementIntent::Idle;
+//    }
+//}
 
-        assert_eq!(expected_transform, actual_transform);
+pub fn interact_entity(
+    input: Res<ButtonInput<KeyCode>>,
+    mut interactive_event_wrier: EventWriter<PlayerInteraction>,
+) {
+    if !input.just_pressed(KeyCode::KeyE) {
+        return;
     }
 
-    #[test]
-    fn out_of_bounds_bottom() {
-        let mut app = setup_app_bounds_checking();
-
-        let player_id = app
-            .world
-            .spawn_empty()
-            .insert((
-                Player,
-                Transform::from_xyz(
-                    TEST_LEVEL_WIDTH_IN_BOUNDS,
-                    TEST_LEVEL_HEIGHT_OUT_BBOUNDS,
-                    0.0,
-                ),
-            ))
-            .id();
-
-        app.update();
-
-        let player_query = app.world.get::<Transform>(player_id);
-        assert!(player_query.is_some());
-
-        let expected_transform =
-            Transform::from_xyz(TEST_LEVEL_WIDTH_IN_BOUNDS, PLAYER_MIDPOINT as f32, 0.0);
-        let actual_transform =
-            *player_query.expect("out_of_bounds_bottom [test]: Player could not be found");
-
-        assert_eq!(expected_transform, actual_transform);
-    }
+    interactive_event_wrier.send(PlayerInteraction);
 }
+
+// pub fn display_interactive_message(mut interactible_event_reader: EventReader<InteractionEvent>) {
+//     for interaction_command in interactible_event_reader.read() {
+//         let command = &interaction_command.0;
+//         if command != "message" {
+//             continue;
+//         }
+
+//         let arg = &interaction_command.1;
+//         println!("{}", arg);
+//     }
+// }
```

Comment:
I recommend removing this as well, since we'll be dealing with this in a future feature.

Reply:
I agree, there's nothing we'd want to salvage from this function